### ISSUE TITLE
Omits loop bodies without adding unimplemented!()

### ISSUE THREAD

**Issue by burdges** [immunant/c2rust#331]
If I enter into https://c2rust.com/ the code
```
uint32_t bitpoly_mul16(uint32_t a , uint32_t b )
{
 uint32_t r =0;
 for(int i=0;i<16;i++) {
   if( (b>>i)&1 ) r ^= (a<<i);
 }
 return r;
}

uint32_t gf_mul_0x1002d(uint32_t a,uint32_t b)
{
 int len = 16;
 uint32_t r = bitpoly_mul16(a,b);
 uint32_t red = 0x1002d;
 for(int i=len*2-1;i>=len;i--) {
   if( r&(1<<i) ) r ^= (red<<(i-len));
 }
 return r;
}
```
then it produces
```
#[no_mangle]
pub unsafe extern "C" fn bitpoly_mul16(mut a: libc::c_int, mut b: libc::c_int)
 -> libc::c_int {
    let mut i: libc::c_int = 0 as libc::c_int;
    while i < 16 as libc::c_int { i += 1 }
    panic!("Reached end of non-void function without returning");
}
#[no_mangle]
pub unsafe extern "C" fn gf_mul_0x1002d(mut a: libc::c_int,
                                        mut b: libc::c_int) -> libc::c_int {
    let mut len: libc::c_int = 16 as libc::c_int;
    let mut i: libc::c_int = len * 2 as libc::c_int - 1 as libc::c_int;
    while i >= len { i -= 1 }
    panic!("Reached end of non-void function without returning");
}
```
which omits loop bodies entirely.

If c2rust cannot translate the C syntax tree then it should ideally insert `unimplemented!()`s wherever it stops working, or maybe warn the users otherwise. 

Apologies if I missed some documentation about this somewhere, but I just assumed arithmetic would be translated correctly.  I suppose c2rust might only be meant to translate item type declarations, not bodies.

**Comment by thedataking** [comment]
Thanks for the report; this looks like a bug at first glance. The version running on c2rust.com is quite old. Can you try downloading master and see if the problem persists. Try these parameters `--disable-refactoring` and `--no-incremental-relooper`. My best guess is that you're hitting a bug related to the loop transformation code (relooper).

**Comment by burdges** [comment]
This was a quick one off, but I'll do something larger in a few days, so maybe I can explore master then.   Thanks!

**Comment by burdges** [comment]
I've explored some issues here for build advice, but ultimately always hit "extern crate rustc;"  "can't find crate" on Mac OS.  I've only begun trying builds on linux, nothing worked yet, but also not tried so hard yet.

Just fyi, I'm trying to translate the header files in https://github.com/w3f/rs-ec-perf/commit/26045a9eaf92019ebfb56b7935de1d310f2fbffd which implement GF(16) and GF(256) using SIMD and non-SIMD arithmetic.  The original working C code lives in the gf16*.h files in https://github.com/fast-crypto-lab/rainbow-submission-round2/tree/master/Alternative_Implementation/ssse3/IIIc_Classic and I've already ported the tables since that was trivial, just preserving the alignment.  I'd assume the SIMD must all be ported manually too, but that doing so is mostly trivial `%s/../../g` operations.  

I've noticed rust gives `as` an extremely high precedence, but the human eye does not really recogonize this, so often its better to add unnecessary parentheses like `(foo as .. as ..) << ..`.  As c2rust necessarily adds some `as` doing this might make the code more readable.
